### PR TITLE
Feature/game flow

### PR DESCRIPTION
**Pull request by DanielNijkamp**

### CODE REVIEW

**Comment by DanielNijkamp** [comment]
Only files to be added: Room spawning script/game state manager

File: Assets/Scripts/Framework/SceneLoader.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+public class SceneLoader : MonoBehaviour
```
Comment (Kazuma-47) [inline Assets/Scripts/Framework/SceneLoader.cs]:
```suggestion
public sealed class SceneLoader : MonoBehaviour
```

File: Assets/Scripts/Gameflow/Corridor.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    private void CheckPassage()
+    {
+        if (allowPassage && playerCount == playerManager.PlayerCount)
+        {
+            onPassage?.Invoke();
+        }
+    }
+    
+    
+
```
Comment (Kazuma-47) [inline Assets/Scripts/Gameflow/Corridor.cs]:
```suggestion
```

File: Assets/Scripts/UI/Main Menu/MenuHandler.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class MainMenuHandler : MonoBehaviour
```
Comment (Kazuma-47) [inline Assets/Scripts/UI/Main Menu/MenuHandler.cs]:
```suggestion
public  sealed class MainMenuHandler : MonoBehaviour
```

**Review by Kazuma-47 — CHANGES_REQUESTED** [review]

**Review by Kazuma-47 — DISMISSED** [review]
lookin good

**Review by Kazuma-47 — APPROVED** [review]

**Pull request merged**